ExampleMod/Content/Tiles/ExampleHerb.cs

```diff
@@ -58,7 +58,8 @@ public class ExampleHerb : ModTile
 			};
 			TileObjectData.newTile.AnchorAlternateTiles = new int[] {
 				TileID.ClayPot,
-				TileID.PlanterBox
+				TileID.PlanterBox,
+				ModContent.TileType<ExamplePlanterBox>()
 			};
```

Comment:
This is a bit tricky because ideally it should include _all_ modded planter boxes, not just vanillas + the one from the same mod. I'm not sure what the solution here would be except manually change the tileobjectdata somewhere around `TileLoader.PostSetupTileMerge` (when every mod has populated `TileID.Sets.PlanterBoxes`) for every modded tile that has `TileID.PlanterBox` in its `AnchorAlternateTiles`

Reply:
It would require redoing everything that uses AnchorAlternateTiles or making a wrapper class that overrides equality comparison with ints, but making AnchorAlternateTiles support sets (or adding a separate field for sets) would solve this issue and potentially future ones.